software/src/8.0/src/kernel/VkUUID.cpp

```diff
@@ -58,16 +31,43 @@ typedef caddr_t ifc_buf_t;
  *******************************************
  *******************************************/
 
+typedef V::uuid_t::u32	u32;
+typedef V::uuid_t::u16	u16;
+typedef V::uuid_t::u8	u8;
+
 /***********************************
  *****  DCE Compliant Systems  *****
  ***********************************/
 
-#if defined(_AIX) || defined(__hpux)
-#define DCE_UUIDGEN
+#if defined(_WIN32)
+#define USING_WINDOWS_DCE
+#include <rpcdce.h>
+#pragma comment(lib, "rpcrt4.lib")
 
-typedef uuid_t::u32	u32;
-typedef uuid_t::u16	u16;
-typedef uuid_t::u8	u8;
+namespace {
+    uuid_t* UCast (V::uuid_t const &rUUID) {
+	return (uuid_t*)(&rUUID);
+    };
+    uuid_t* UCast (V::uuid_t &rUUID) {
+	return (uuid_t*)(&rUUID);
+    };
+}
+
+#elif defined(__APPLE__) || defined(__linux__) || defined(sun)
+#define USING_LIBUUID_DCE
+#include <uuid/uuid.h>
```

Comment:
This isn't necessarily builtin on Linux:

```
../kernel/VkUUID.cpp:58:23: fatal error: uuid/uuid.h: No such file or directory
 #include <uuid/uuid.h>
```

is it your intention to make libuuid-devel a requirement?